From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: PadowYT2 <me@padow.ru>
Date: Sun, 29 Dec 2024 00:26:54 +0300
Subject: [PATCH] Add padow plugins


diff --git a/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java b/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java
index 865dc183276720d54d31d2a54d1bb5c845e80598..7bd58b563c8a04347f0a8818633ef5d77540b190 100644
--- a/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java
+++ b/src/main/java/io/papermc/paper/command/PaperPluginsCommand.java
@@ -63,6 +63,7 @@ public class PaperPluginsCommand extends BukkitCommand {
 
     private static final Component LEGACY_PLUGIN_STAR = Component.text('*', TextColor.color(255, 212, 42)).hoverEvent(LEGACY_PLUGIN_INFO);
     private static final Component INFO_ICON_START = Component.text("ℹ ", INFO_COLOR);
+    private static final Component PADOW_HEADER = Component.text("Padow Plugins:", TextColor.color(37, 96, 145));
     private static final Component PAPER_HEADER = Component.text("Paper Plugins:", TextColor.color(2, 136, 209));
     private static final Component BUKKIT_HEADER = Component.text("Bukkit Plugins:", TextColor.color(237, 129, 6));
     private static final Component PLUGIN_TICK = Component.text("- ", NamedTextColor.DARK_GRAY);
@@ -220,6 +221,7 @@ public class PaperPluginsCommand extends BukkitCommand {
     public boolean execute(@NotNull CommandSender sender, @NotNull String currentAlias, @NotNull String[] args) {
         if (!this.testPermission(sender)) return true;
 
+        TreeMap<String, PluginProvider<JavaPlugin>> padowPlugins = new TreeMap<>(String.CASE_INSENSITIVE_ORDER); // Padow
         TreeMap<String, PluginProvider<JavaPlugin>> paperPlugins = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
         TreeMap<String, PluginProvider<JavaPlugin>> spigotPlugins = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
 
@@ -228,7 +230,13 @@ public class PaperPluginsCommand extends BukkitCommand {
             PluginMeta configuration = provider.getMeta();
 
             if (provider instanceof SpigotPluginProvider) {
-                spigotPlugins.put(configuration.getDisplayName(), provider);
+                // Padow
+                if (configuration.getAuthors().contains("PadowYT2")) {
+                    padowPlugins.put(configuration.getDisplayName(), provider);
+                } else {
+                    spigotPlugins.put(configuration.getDisplayName(), provider);
+                }
+                // Padow
             } else if (provider instanceof PaperPluginParent.PaperServerPluginProvider) {
                 paperPlugins.put(configuration.getDisplayName(), provider);
             }
@@ -239,6 +247,14 @@ public class PaperPluginsCommand extends BukkitCommand {
 
         //sender.sendMessage(infoMessage); // Purpur
 
+        // Padow
+        sender.sendMessage(PADOW_HEADER.append(Component.text(" (%s):".formatted(padowPlugins.size())))); // Purpur
+
+        for (Component component : formatProviders(padowPlugins, sender)) { // Purpur
+            sender.sendMessage(component);
+        }
+        // Padow
+
         //if (!paperPlugins.isEmpty()) { // Purpur
             sender.sendMessage(PAPER_HEADER.append(Component.text(" (%s):".formatted(paperPlugins.size())))); // Purpur
         //} // Purpur
